pages/college_predictor.js

```diff
@@ -112,63 +153,83 @@ const CollegePredictor = () => {
   }, [fetchAndFilterData]);
 
   return (
-    <div className="flex flex-col items-center p-4">
-      <div className="flex flex-col items-center justify-center w-full sm:w-5/6 md:w-3/4 lg:w-2/3 bg-white p-6 rounded-lg shadow-lg">
-        <h1 className="text-2xl font-bold mb-4 text-center">
-          {getConstants().TITLE}
-        </h1>
-        <div className="text-center mb-4 space-y-2">
-          <p className="text-base md:text-lg">
-            {query.exam !== "NEET"
-              ? `Your Category Rank: ${query.rank}`
-              : `Your Rank: ${query.rank}`}
-          </p>
-          {query.exam !== "MHT CET" && query.exam !== "KCET" && (
-            <p className="text-base md:text-lg">
-              Chosen Round Number: {query.roundNumber}
-            </p>
-          )}
-          <p className="text-base md:text-lg">Chosen Exam: {query.exam}</p>
-          {query.exam !== "NEET" && query.exam !== "KCET" && (
+    <>
+      <Head>
+        <title>College Predictor - Result</title>
+      </Head>
+      <div className="flex flex-col items-center p-4">
+        <div className="flex flex-col items-center justify-center w-full sm:w-5/6 md:w-3/4 lg:w-2/3 bg-white p-6 rounded-lg shadow-lg">
+          <h1 className="text-2xl font-bold mb-4 text-center">
+            {getConstants().TITLE}
+          </h1>
+          <div className="text-center mb-4 space-y-2">
             <p className="text-base md:text-lg">
-              Chosen Gender: {query.gender}
+              {query.exam !== "NEET"
+                ? `Your Category Rank: ${query.rank}`
+                : `Your Rank: ${query.rank}`}
             </p>
-          )}
-          {query.exam !== "NEET" && (
-            <p className="text-base md:text-lg">
-              Chosen Home State: {query.stateName}
-            </p>
-          )}
-          {query.exam === "KCET" && (
-            <>
+            {query.exam !== "MHT CET" && query.exam !== "KCET" && (
```

Comment:
again.. this isn't good code either. we can make a general display here. whatever filters are possible for a particular exam, iterate over all of them and show